files/image_config/rsyslog/rsyslog-config.sh

```diff
@@ -1,4 +1,24 @@
 #!/bin/bash
 
-sonic-cfggen -d -t /usr/share/sonic/templates/rsyslog.conf.j2 >/etc/rsyslog.conf
+PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+
+# Parse the device specific asic conf file, if it exists
+ASIC_CONF=/usr/share/sonic/device/$PLATFORM/asic.conf
+if [ -f "$ASIC_CONF" ]; then
+    source $ASIC_CONF
+fi
+
+# On Multi NPU platforms we need to start  the rsyslog server on the docker0 ip address
+# for the syslogs from the containers in the namespaces to work.
+# on Single NPU platforms we continue to use loopback adddres
+
+if [[ ($NUM_ASIC -gt 1) ]]; then
+    intf="docker0"
+else
+    intf="lo"
+fi
+
+udp_server_ip=$(ip -o -4 addr list $intf | awk '{print $4}' | cut -d/ -f1)
```

Comment:
>ip -o -4 addr list $intf [](start = 16, length = 24)

There may be more than one lo interfaces. How do you handle?

Reply:
fixed in the latest commit